packages/kbn-test/src/failed_tests_reporter/report_failures_to_file.ts

```diff
@@ -0,0 +1,141 @@
+    if (statSync(join(dirPath, file)).isDirectory()) {
+      if (file.match(/node_modules/)) {
+        continue;
+      }
+
+      allScreenshots = findScreenshots(join(dirPath, file), allScreenshots);
+    } else {
+      const fullPath = join(dirPath, file);
+      if (fullPath.match(/screenshots\/failure\/.+\.png$/)) {
+        allScreenshots.push(fullPath);
+      }
+    }
+  }
+
+  return allScreenshots;
+};
+
+export function reportFailuresToFile(log: ToolingLog, failures: TestFailure[]) {
+  if (!failures?.length) {
+    return;
+  }
+
+  let screenshots: string[];
+  try {
+    screenshots = [
+      ...findScreenshots(join(REPO_ROOT, 'test', 'functional')),
+      ...findScreenshots(join(REPO_ROOT, 'x-pack', 'test', 'functional')),
+    ];
+  } catch (e) {
+    log.error(e as Error);
+    screenshots = [];
+  }
+
+  const screenshotsByName: Record<string, string> = {};
+  for (const screenshot of screenshots) {
+    const [name] = basename(screenshot).split('.');
+    screenshotsByName[name] = screenshot;
+  }
+
+  // Jest could, in theory, fail 1000s of tests and write 1000s of failures
+  // So let's just write files for the first 20
+  for (const failure of failures.slice(0, 20)) {
+    const hash = createHash('md5').update(failure.name).digest('hex');
+    const filenameBase = `${
+      process.env.BUILDKITE_JOB_ID ? process.env.BUILDKITE_JOB_ID + '_' : ''
+    }${hash}`;
+    const dir = join('target', 'test_failures');
+
+    const failureLog = [
+      ['Test:', '-----', failure.classname, failure.name, ''],
+      ['Failure:', '--------', failure.failure],
+      failure['system-out'] ? ['', 'Standard Out:', '-------------', failure['system-out']] : [],
+    ]
+      .flat()
+      .join('\n');
+
+    const failureJSON = JSON.stringify(
+      {
+        ...failure,
+        hash,
+        buildId: process.env.BUJILDKITE_BUILD_ID || '',
+        jobId: process.env.BUILDKITE_JOB_ID || '',
+        url: process.env.BUILDKITE_BUILD_URL || '',
+        jobName: process.env.BUILDKITE_LABEL
+          ? `${process.env.BUILDKITE_LABEL}${
+              process.env.BUILDKITE_PARALLEL_JOB ? ` #${process.env.BUILDKITE_PARALLEL_JOB}` : ''
+            }`
+          : '',
+      },
+      null,
+      2
+    );
+
+    let screenshot = '';
+    const screenshotName = `${failure.name.replace(/([^ a-zA-Z0-9-]+)/g, '_')}`;
+    if (screenshotsByName[screenshotName]) {
+      try {
+        screenshot = readFileSync(screenshotsByName[screenshotName]).toString('base64');
+      } catch (e) {
+        log.error(e as Error);
+      }
+    }
+
+    const screenshotHtml = screenshot
+      ? `<img class="screenshot img-fluid img-thumbnail" src="data:image/png;base64,${screenshot}" />`
+      : '';
+
+    const failureHTML = readFileSync(
+      join(
+        REPO_ROOT,
+        'packages',
+        'kbn-test',
+        'src',
+        'failed_tests_reporter',
+        'report_failures_to_file_html_template.html'
+      )
```

Comment:
It would be a lot more compact to use, this is one of the reasons I prefer `resolve()` to `join()`:
```suggestion
      Path.resolve(
        REPO_ROOT,
        'packages/kbn-test/src/failed_tests_reporter/report_failures_to_file_html_template.html'
      )
```